pennylane/operation.py

```diff
@@ -376,9 +522,12 @@ def _eigvals(cls, *params):
         array([1, -1])
 
         Returns:
-            array: eigenvalue representation
+            array: eigenvalue representation or None
         """
-        return np.linalg.eigvals(cls._matrix(*params))
+        mat = cls.compute_matrix(*params)
+        if mat is None:
+            return None
```

Comment:
Need to propagate the possible case of an undefined matrix here!

Reply:
Ohhh actually I guess that was quite nice in the old system --- you didn't need to worry about propagating `None`s, because a single `NotImplementedError` would break the chain for you automatically

Reply:
Yes that's true! But then, we never knew where that error came from, so it was more opaque as well.